upgrade/upgrade-scripts/rootfs-container

```diff
@@ -77,7 +77,14 @@ function get_bootloader_devices() {
 	#
 	zpool list -vH rpool |
 		awk '! /rpool|mirror|replacing|spare/ {print $1}' |
-		sed 's/p\{0,1\}[0-9]*$//'
+		while read -r part; do
+			# If it doesn't end with a number, it's already a full disk
+			if [[ "$part" =~ [a-z]$ ]]; then
+				echo "$part"
```

Comment:
presumably this is done because `lsblk` doesn't work when the input is not a partition?

IMO, rather than trying to support this case, I think it might be better to throw an error. I say that because for our (non-migrated) Linux systems, rpool will always use a partition, so this code path won't ever be tested/used. We'd need to verify this for migrated systems.

What do you think? I'm just hesitant to add a branch that'll never get exercised.

Reply:
I think that makes sense, I'll switch this over to throwing an error.

Reply:
Experimenting a little, my suggestion is to unconditionally call:
```
lsblk --nodeps -no pkname "/dev/$part"
```
and then throw an error if that returns an empty string..

For example:
```
$ lsblk --nodeps -no pkname /dev/sda

$ lsblk --nodeps -no pkname /dev/sda1
sda
```

I'm assuming a migrated system also uses partitions for rpool, just like a non-migrated system. If a migrated system does not use partitions for rpool, then perhaps we could instead use `--nodeps -no type` to determine if it's a partition or not (rather than simply checking the string name):
```
$ lsblk --nodeps -no type /dev/sda1
part
$ lsblk --nodeps -no type /dev/sda
disk
```

Reply:
I checked on a migrated VM, our root pool is also installed on a partition in that case.